core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/common/iterator/AbstractOriginalSwapIterator.java

```diff
@@ -40,17 +40,14 @@ public AbstractOriginalSwapIterator(ListIterable<SubSelection_> leftSubSelector,
 
     @Override
     protected Move_ createUpcomingSelection() {
-        if (!rightSubSelectionIterator.hasNext()) {
+        while (!rightSubSelectionIterator.hasNext()) {
```

Comment:
Why is this change only being made to the swap iterator?
Do no other moves suffer from this problem?

Reply:
This issue only affects the original iterators. The change move selector already uses a while loop. However, it seems to also impact `OriginalPillarChangeMoveIterator`. I'll add a test to verify this.